core/src/main/java/com/dtolabs/rundeck/plugins/webhook/WebhookEventPlugin.java

```diff
@@ -25,5 +25,5 @@
  */
 public interface WebhookEventPlugin {
     default List<String> getRequestHeadersToCopy() {return new ArrayList<>(); };
-    void onEvent(WebhookEventContext context, WebhookData data) throws WebhookEventException;
+    WebhookResponder onEvent(WebhookEventContext context, WebhookData data) throws WebhookEventException;
```

Comment:
i think to preserve compatibility, this interface method shouldn't change.  perhaps add another method that returns WebhookResponder type that has a default implementation so older implementations still work e.g. `default WebhookResponder handleEvent(...){ return null; }`. to differentiate which method to use, there could be another method `default boolean handlesEvent(){ return false; }`

Reply:
I went back and forth on maintaining backwards compatibility. 

I think it would be better to take advantage of the 3.3.x version bump and document the breaking change, rather than increase the complexity of the API and pollute it with deprecated methods that should not be used. My thought is that since the webhook API is relatively new, the need to maintain backward compatibility is less urgent.

Reply:
ok, if we add the WebhookResponder return type to the method, then it would be nice if existing implementations could just update their methods to return `null`, and have the same default response be sent as it was before.  I think the WebhookService or controller would have to change to handle that